tornado-unittests/src/main/java/uk/ac/manchester/tornado/unittests/compute/LLMFusedKernelsTest.java

```diff
@@ -0,0 +1,286 @@
+            float eps = 1e-5f;
+
+
+            FloatArray x = new FloatArray(size);
+            FloatArray state = new FloatArray(size);
+            FloatArray rnsOutput = new FloatArray(size);
+            FloatArray sequentialOutput;
+
+            FloatArray weights = new FloatArray(size * size);
+            FloatArray outputLogits = new FloatArray(size);
+            FloatArray outputSeqLogits = new FloatArray( size);
+
+            FloatArray reduce = new FloatArray(size / localSize);
+
+            outputLogits.init(0f);
+            outputSeqLogits.init(0f);
+
+            for (int i = 0; i < size; i++) {
+                x.set(i, (float) (i + 1));
+                state.set(i, 0.5f+i);
+            }
+
+            for (int i = 0; i < size * size; i++) {
+                weights.set(i, 0.8f+i);
+            }
+
+            sequentialOutput = rmsnorm(x, weights, size, eps);
+            outputSeqLogits = matmul(weights, sequentialOutput, outputSeqLogits, size, size);
+
+
+            KernelContext context = new KernelContext();
+            TaskGraph taskGraph = new TaskGraph("fused") //
+                .transferToDevice(DataTransferMode.FIRST_EXECUTION, size, eps, x, reduce, weights) //
+                .transferToDevice(DataTransferMode.EVERY_EXECUTION, localSize, state) //
+                .task("reduce", LLMFusedKernelsTest::reduceSquareSums, context, x, reduce) //
+                .task("sum", LLMFusedKernelsTest::finalSum, context, reduce, size, eps) //
+                .task("ns", LLMFusedKernelsTest::normalizeAndScale, context, rnsOutput, x, state, reduce, size, eps) //
+                .task("mv", LLMFusedKernelsTest::matrixVector, context, weights, rnsOutput, outputLogits, size, size) //
+                .transferToHost(DataTransferMode.EVERY_EXECUTION, outputLogits); //
+
+
+            WorkerGrid worker = new WorkerGrid1D(size);
+            worker.setGlobalWork(size, 1, 1);
+            worker.setLocalWork(localSize, 1, 1);
+            GridScheduler gridScheduler = new GridScheduler("fused.reduce", worker);
+            gridScheduler.setWorkerGrid("fused.sum", new WorkerGrid1D(1));
+            gridScheduler.setWorkerGrid("fused.ns", worker);
+            gridScheduler.setWorkerGrid("fused.mv", worker);
+
+            ImmutableTaskGraph immutableTaskGraph = taskGraph.snapshot();
+            try (TornadoExecutionPlan executionPlan = new TornadoExecutionPlan(immutableTaskGraph)) {
+                executionPlan.withGridScheduler(gridScheduler) //
+                        .execute();
+            }
+
+
+            for (int i = 0; i < outputSeqLogits.getSize(); i++) {
+                float expected = outputSeqLogits.get(i);           // Expected value from the sequential output
+                float actual = outputLogits.get(i);               // Actual value from the RNS output
+
+//                assertEquals("Mismatch at index " + i, expected, actual, 1f); // Allow some tolerance
```

Comment:
Remove comment

Reply:
I guess the test should assert the expected and actual values, right?